modules/social_features/social_group/modules/social_group_request/src/Plugin/Block/SocialGroupRequestMembershipNotification.php

```diff
@@ -0,0 +1,159 @@
+ * )
+ */
+class SocialGroupRequestMembershipNotification extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * User account entity.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $account;
+
+  /**
+   * Group entity.
+   *
+   * @var \Drupal\group\Entity\GroupInterface
+   */
+  protected $group;
+
+  /**
+   * Entity type manger.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs SocialGroupRequestMembershipNotification.
+   *
+   * @param array $configuration
+   *   Configuration array.
+   * @param string $plugin_id
+   *   Plugin ID.
+   * @param mixed $plugin_definition
+   *   Plugin definition.
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   User account entity.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, AccountInterface $account, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->account = $account;
+    $this->group = _social_group_get_current_group();
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('current_user'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    if (!$this->group->getGroupType()->hasContentPlugin('group_membership_request')) {
+      return [];
+    }
+
+    if ($this->group->getGroupType()->id() === 'flexible_group') {
+      $join_methods = $this->group->get('field_group_allowed_join_method')->getValue();
+      $request_option = in_array('request', array_column($join_methods, 'value'), FALSE);
+      if (!$request_option) {
+        return [];
+      }
+    }
+    else {
+      $allow_request = $this->group->get('allow_request');
+      if ($allow_request->isEmpty() || $allow_request->value == 0) {
+        return [];
+      }
+    }
+
+    $contentTypeConfigId = $this->group
+      ->getGroupType()
+      ->getContentPlugin('group_membership_request')
+      ->getContentTypeConfigId();
+
+    $requests = $this->entityTypeManager->getStorage('group_content')->getQuery()
+      ->condition('type', $contentTypeConfigId)
+      ->condition('gid', $this->group->id())
+      ->condition('grequest_status', GroupMembershipRequest::REQUEST_PENDING)
+      ->count()
+      ->execute();
+
+    if (!$requests) {
+      return [];
+    }
+
+    return [
+      '#type' => 'html_tag',
+      '#tag' => 'div',
+      '#value' => $this->t('There are @link to join this group.', [
```

Comment:
We should split this up and use `formatPlural`. It'll be a bit harder as you need to check on multiple things, but it will look much better when there is just 1 request (which is often the case).

You could check how they did this at the Request to Join Event feature: https://github.com/goalgorilla/open_social/pull/1756/files#diff-7fdc86930796f1d8427e7de6935cefdbR591.

Reply:
Good point! Fixed

Reply:
@denysyukm almost there!

Maybe you can include `are` in `(@count) new requests` and `is` at `(1) new request)`. That will make it look great.